src/sentry/users/api/endpoints/user_details.py

```diff
@@ -39,6 +39,13 @@
 TIMEZONE_CHOICES = get_timezone_choices()
 
 
+def validate_prefers_specialized_project_overview(value: dict[str, bool] | None) -> None:
+    if value is not None:
+        for entry in value.values():
+            if not isinstance(entry, bool):
+                raise ValidationError("The enabled value should be a boolean.")
+
+
```

Comment:
This would only report the first non-boolean value.
```suggestion
def validate_prefers_specialized_project_overview(value: dict[str, bool] | None) -> None:
    invalid_entries = [
        key
        for key, value_ in (value or {}).items()
        if not isinstance(value_, bool)
    ]
    raise ValidationError(f"The enabled values {', '.join(invalid_entries)} should be booleans.")

```